include/usearch/index.hpp

```diff
@@ -3455,8 +3641,16 @@ class index_gt {
             return result.failed("File is corrupted and lacks a header");
         std::memcpy(&header, file.data() + offset, sizeof(header));
 
+        // Keep offset for re-using later.
+        mapped_file_offset_ = offset;
+        // Viewed index is immutable.
+        is_mutable_ = false;
+
         if (!header.size) {
-            reset();
```

Comment:
We don't need to call `reset()` because we don't change anything after the above `reset()`.